<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Repeat CRC paper's Figure 4A • PCAGenomicSignaturesPaper</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Repeat CRC paper's Figure 4A">
<meta property="og:description" content="PCAGenomicSignaturesPaper">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">PCAGenomicSignaturesPaper</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    CRC
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/CRC/PCSS_vs_RAV.html">RAVs similar to PCSS1/2</a>
    </li>
    <li>
      <a href="../../articles/CRC/metadata_based_search.html">Metadata based search</a>
    </li>
    <li>
      <a href="../../articles/CRC/repeat_Fig4A.html">CRC subtype scores</a>
    </li>
    <li>
      <a href="../../articles/CRC/repeat_Fig4C.html">CRC clinical variables</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    multiPLIER
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/SLE-WB/repeat_Fig3_SLE-WB.html">Neutrophil (SLE-WB)</a>
    </li>
    <li>
      <a href="../../articles/NARES/repeat_Fig3_NARES.html">Neutrophil (NARES)</a>
    </li>
    <li>
      <a href="../../articles/others/repeat_Fig5_Pathway_Separation.html">Pathway separation</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/shbrief/PCAGenomicSignaturesPaper">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="repeat_Fig4A_files/accessible-code-block-0.0.1/empty-anchor.js"></script><link href="repeat_Fig4A_files/anchor-sections-1.0/anchor-sections.css" rel="stylesheet">
<script src="repeat_Fig4A_files/anchor-sections-1.0/anchor-sections.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Repeat CRC paper’s Figure 4A</h1>
                        <h4 class="author">Sehyun Oh</h4>
            
            <h4 class="date">2020-11-02</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/shbrief/PCAGenomicSignaturesPaper/blob/master/vignettes/CRC/repeat_Fig4A.Rmd"><code>vignettes/CRC/repeat_Fig4A.Rmd</code></a></small>
      <div class="hidden name"><code>repeat_Fig4A.Rmd</code></div>

    </div>

    
    
<div id="setup" class="section level1">
<h1 class="hasAnchor">
<a href="#setup" class="anchor"></a>Setup</h1>
<p>In this vignette, we are reproducing Figure 4A of the <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-018-1511-4">CRC paper</a> using PCAGenomicSignatures.</p>
<div id="pcamodel" class="section level2">
<h2 class="hasAnchor">
<a href="#pcamodel" class="anchor"></a>PCAmodel</h2>
<pre><code>## class: PCAGenomicSignatures 
## dim: 13934 4764 
## metadata(6): cluster size ... MeSH_freq updateNote
## assays(1): model
## rownames(13934): CASKIN1 DDX3Y ... CTC-457E21.9 AC007966.1
## rowData names(0):
## colnames(4764): RAV1 RAV2 ... RAV4763 RAV4764
## colData names(4): RAV studies silhouetteWidth gsea
## trainingData(2): PCAsummary MeSH
## trainingData names(536): DRP000987 SRP059172 ... SRP164913 SRP188526
## [1] "536 refine.bio studies/ top 90% varying genes/ GSEA with MSigDB C2"</code></pre>
</div>
<div id="crc-validation-datasets" class="section level2">
<h2 class="hasAnchor">
<a href="#crc-validation-datasets" class="anchor"></a>CRC validation datasets</h2>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"data/eSets/setNames.RData"</span><span class="op">)</span></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="co">## Load validation samples</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">set</span> <span class="kw">in</span> <span class="va">setNames</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"data/eSets_new/"</span>, <span class="va">set</span>, <span class="st">'.RData'</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></pre></div>
</div>
</div>
<div id="by-pcss1pcss2" class="section level1">
<h1 class="hasAnchor">
<a href="#by-pcss1pcss2" class="anchor"></a>By PCSS1/PCSS2</h1>
<p>First, we simply reproduced the Figure 4A using PCSS1 and PCSS2.</p>
<p><img src="repeat_Fig4A_files/figure-html/unnamed-chunk-6-1.png" width="80%" height="80%"></p>
</div>
<div id="by-pcagenomicsignatures" class="section level1">
<h1 class="hasAnchor">
<a href="#by-pcagenomicsignatures" class="anchor"></a>By PCAGenomicSignatures</h1>
<div id="most-similar-to-pcss1pcss2" class="section level2">
<h2 class="hasAnchor">
<a href="#most-similar-to-pcss1pcss2" class="anchor"></a>Most similar to PCSS1/PCSS2</h2>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="va">sampleScore1</span> <span class="op">&lt;-</span> <span class="fl">1575</span>
<span class="va">sampleScore2</span> <span class="op">&lt;-</span> <span class="fl">834</span>
<span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span><span class="op">(</span><span class="st">"R/Fig4A_plotting.R"</span>, print.eval <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></pre></div>
<p><img src="repeat_Fig4A_files/figure-html/unnamed-chunk-8-1.png" width="80%" height="80%"></p>
<pre><code>## `summarise()` regrouping output by 'study' (override with `.groups` argument)</code></pre>
<p><img src="repeat_Fig4A_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<div id="wordcloud" class="section level3">
<h3 class="hasAnchor">
<a href="#wordcloud" class="anchor"></a>wordcloud</h3>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="fu">drawWordcloud</span><span class="op">(</span><span class="va">PCAmodel</span>, <span class="va">sampleScore1</span><span class="op">)</span>
<span class="co">## Warning in wordcloud::wordcloud(words = all$word, freq = all$freq, scale =</span>
<span class="co">## scale, : Drug Resistance, Neoplasm could not be fit on page. It will not be</span>
<span class="co">## plotted.</span>
<span class="co">## Warning in wordcloud::wordcloud(words = all$word, freq = all$freq, scale =</span>
<span class="co">## scale, : Antigen Presentation could not be fit on page. It will not be plotted.</span>
<span class="co">## Warning in wordcloud::wordcloud(words = all$word, freq = all$freq, scale =</span>
<span class="co">## scale, : Carcinoma, Hepatocellular could not be fit on page. It will not be</span>
<span class="co">## plotted.</span></pre></div>
<p><img src="repeat_Fig4A_files/figure-html/unnamed-chunk-10-1.png" width="60%" height="60%"></p>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="fu">drawWordcloud</span><span class="op">(</span><span class="va">PCAmodel</span>, <span class="va">sampleScore2</span><span class="op">)</span>
<span class="co">## Warning in wordcloud::wordcloud(words = all$word, freq = all$freq, scale =</span>
<span class="co">## scale, : Histocompatibility Testing could not be fit on page. It will not be</span>
<span class="co">## plotted.</span>
<span class="co">## Warning in wordcloud::wordcloud(words = all$word, freq = all$freq, scale =</span>
<span class="co">## scale, : Massively Parallel Sequencing could not be fit on page. It will not be</span>
<span class="co">## plotted.</span>
<span class="co">## Warning in wordcloud::wordcloud(words = all$word, freq = all$freq, scale =</span>
<span class="co">## scale, : Histocompatibility Antigens Class II could not be fit on page. It will</span>
<span class="co">## not be plotted.</span></pre></div>
<p><img src="repeat_Fig4A_files/figure-html/unnamed-chunk-10-2.png" width="60%" height="60%"></p>
</div>
<div id="gsea" class="section level3">
<h3 class="hasAnchor">
<a href="#gsea" class="anchor"></a>GSEA</h3>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="fu">annotateRAV</span><span class="op">(</span><span class="va">PCAmodel</span>, <span class="va">sampleScore1</span><span class="op">)</span>
<span class="co">##                                  Description      NES pvalue      qvalues</span>
<span class="co">## 1   SCHUETZ_BREAST_CANCER_DUCTAL_INVASIVE_UP 2.794522  1e-10 1.733483e-09</span>
<span class="co">## 2          SMID_BREAST_CANCER_NORMAL_LIKE_UP 2.681439  1e-10 1.733483e-09</span>
<span class="co">## 3                       BOQUEST_STEM_CELL_UP 2.630321  1e-10 1.733483e-09</span>
<span class="co">## 4 VECCHI_GASTRIC_CANCER_ADVANCED_VS_EARLY_UP 2.596083  1e-10 1.733483e-09</span>
<span class="co">## 5                 MCLACHLAN_DENTAL_CARIES_UP 2.566848  1e-10 1.733483e-09</span>
<span class="fu">annotateRAV</span><span class="op">(</span><span class="va">PCAmodel</span>, <span class="va">sampleScore2</span><span class="op">)</span>
<span class="co">##                                             Description      NES pvalue</span>
<span class="co">## 1 ALTEMEIER_RESPONSE_TO_LPS_WITH_MECHANICAL_VENTILATION 2.868397  1e-10</span>
<span class="co">## 2                            MCLACHLAN_DENTAL_CARIES_UP 2.627526  1e-10</span>
<span class="co">## 3                           WIELAND_UP_BY_HBV_INFECTION 2.541545  1e-10</span>
<span class="co">## 4            ZHANG_RESPONSE_TO_IKK_INHIBITOR_AND_TNF_UP 2.339746  1e-10</span>
<span class="co">## 5                       POOLA_INVASIVE_BREAST_CANCER_UP 2.334263  1e-10</span>
<span class="co">##        qvalues</span>
<span class="co">## 1 4.683457e-08</span>
<span class="co">## 2 4.683457e-08</span>
<span class="co">## 3 4.683457e-08</span>
<span class="co">## 4 4.683457e-08</span>
<span class="co">## 5 4.683457e-08</span></pre></div>
</div>
</div>
<div id="best-validation-scores-overall-on-crc-validation-datasets" class="section level2">
<h2 class="hasAnchor">
<a href="#best-validation-scores-overall-on-crc-validation-datasets" class="anchor"></a>Best validation scores overall on CRC validation datasets</h2>
<div id="validation" class="section level3">
<h3 class="hasAnchor">
<a href="#validation" class="anchor"></a>Validation</h3>
<p>Before applying <code>validate</code> function, any NA or Inf values in CRC datasets were removed and the expression matrix was centered.</p>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="va">validated_ind_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html">vector</a></span><span class="op">(</span>mode <span class="op">=</span> <span class="st">"list"</span>, length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">setNames</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">validated_ind_all</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">setNames</span>

<span class="kw">for</span> <span class="op">(</span><span class="va">set</span> <span class="kw">in</span> <span class="va">setNames</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">eSet</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/get.html">get</a></span><span class="op">(</span><span class="va">set</span><span class="op">)</span>
  <span class="va">exprs</span> <span class="op">&lt;-</span> <span class="fu">exprs</span><span class="op">(</span><span class="va">eSet</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">rmNaInf</span>
  <span class="va">exprs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="va">exprs</span>, <span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">t</span>
  <span class="va">val_all</span> <span class="op">&lt;-</span> <span class="fu">validate</span><span class="op">(</span><span class="va">exprs</span>, <span class="va">PCAmodel</span><span class="op">)</span>
  <span class="va">validated_ind_all</span><span class="op">[[</span><span class="va">set</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">validatedSignatures</span><span class="op">(</span><span class="va">val_all</span>, num.out <span class="op">=</span> <span class="fl">10</span>, indexOnly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="op">}</span></pre></div>
<p>We combined the top validated RAVs from all 18 studies and checked the frequency of them. Any RAV that appeared more often means they are consistently associated with the common feature of CRC datasets.</p>
<p>Top row is RAV number and the bottom row is the frequency of it making top validation. Because there were 18 datasets, the maximum frequeny of showing up is 18.</p>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="co">## Top 10 validated</span>
<span class="va">validated_ind_all</span> <span class="op">%&gt;%</span> <span class="va">unlist</span> <span class="op">%&gt;%</span> <span class="va">table</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="va">.</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">## .</span>
<span class="co">##  188  832  833 1575  338  725  324  438    2  312  684 1032   21  192  220  468 </span>
<span class="co">##   14   10    9    9    8    8    6    6    5    5    5    5    4    4    4    4 </span>
<span class="co">##  579 1076 1166 2538  119  187  758  834  884 1467 1694 1843  189  316  504  516 </span>
<span class="co">##    4    4    4    4    3    3    3    3    3    3    3    3    2    2    2    2 </span>
<span class="co">##  595  868  981 1008 1387   27   40   58   61  152  153  170  190 1016 1165 1187 </span>
<span class="co">##    2    2    2    2    2    1    1    1    1    1    1    1    1    1    1    1 </span>
<span class="co">## 1468 1497 1726 2489 2506 </span>
<span class="co">##    1    1    1    1    1</span>

<span class="co">## Top 5 validated</span>
<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">validated_ind_all</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="va">unlist</span> <span class="op">%&gt;%</span> <span class="va">table</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="va">.</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">## .</span>
<span class="co">##  188  338  832  833    2 1575  312  324  725  220  438  684  834 1032 2538   21 </span>
<span class="co">##    9    8    7    6    5    5    4    4    4    3    3    3    3    3    3    2 </span>
<span class="co">##  192  468 1076   58  119  152  189  316  516  579  758  981 1165 1166 1387 </span>
<span class="co">##    2    2    2    1    1    1    1    1    1    1    1    1    1    1    1</span></pre></div>
</div>
<div id="top-validated-ravs" class="section level3">
<h3 class="hasAnchor">
<a href="#top-validated-ravs" class="anchor"></a>Top validated RAVs</h3>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="va">sampleScore1</span> <span class="op">&lt;-</span> <span class="fl">832</span>
<span class="va">sampleScore2</span> <span class="op">&lt;-</span> <span class="fl">188</span>
<span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span><span class="op">(</span><span class="st">"R/Fig4A_plotting.R"</span>, print.eval <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></pre></div>
<p><img src="repeat_Fig4A_files/figure-html/unnamed-chunk-14-1.png" width="80%" height="80%"></p>
<pre><code>## `summarise()` regrouping output by 'study' (override with `.groups` argument)</code></pre>
<p><img src="repeat_Fig4A_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
</div>
<div id="wordcloud-1" class="section level3">
<h3 class="hasAnchor">
<a href="#wordcloud-1" class="anchor"></a>wordcloud</h3>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="fu">drawWordcloud</span><span class="op">(</span><span class="va">PCAmodel</span>, <span class="va">sampleScore1</span><span class="op">)</span>
<span class="co">## Warning in wordcloud::wordcloud(words = all$word, freq = all$freq, scale =</span>
<span class="co">## scale, : Colorectal Cancers could not be fit on page. It will not be plotted.</span></pre></div>
<p><img src="repeat_Fig4A_files/figure-html/unnamed-chunk-17-1.png" width="60%" height="60%"></p>
<div class="sourceCode" id="cb14"><pre class="downlit">
<span class="fu">drawWordcloud</span><span class="op">(</span><span class="va">PCAmodel</span>, <span class="va">sampleScore2</span><span class="op">)</span>
<span class="co">## Warning in wordcloud::wordcloud(words = all$word, freq = all$freq, scale =</span>
<span class="co">## scale, : Transplantation, Autologous could not be fit on page. It will not be</span>
<span class="co">## plotted.</span>
<span class="co">## Warning in wordcloud::wordcloud(words = all$word, freq = all$freq, scale =</span>
<span class="co">## scale, : Inflammatory Bowel Disease could not be fit on page. It will not be</span>
<span class="co">## plotted.</span></pre></div>
<p><img src="repeat_Fig4A_files/figure-html/unnamed-chunk-17-2.png" width="60%" height="60%"></p>
</div>
<div id="gsea-1" class="section level3">
<h3 class="hasAnchor">
<a href="#gsea-1" class="anchor"></a>GSEA</h3>
<div class="sourceCode" id="cb15"><pre class="downlit">
<span class="fu">annotateRAV</span><span class="op">(</span><span class="va">PCAmodel</span>, <span class="va">sampleScore1</span><span class="op">)</span>
<span class="co">##                                                            Description      NES</span>
<span class="co">## 1                                        SABATES_COLORECTAL_ADENOMA_UP 3.086762</span>
<span class="co">## 2                                           HSIAO_LIVER_SPECIFIC_GENES 2.880972</span>
<span class="co">## 3                                     GRADE_COLON_AND_RECTAL_CANCER_UP 2.773660</span>
<span class="co">## 4                           REACTOME_EUKARYOTIC_TRANSLATION_ELONGATION 2.739737</span>
<span class="co">## 5 REACTOME_SRP_DEPENDENT_COTRANSLATIONAL_PROTEIN_TARGETING_TO_MEMBRANE 2.708715</span>
<span class="co">##   pvalue      qvalues</span>
<span class="co">## 1  1e-10 1.059423e-08</span>
<span class="co">## 2  1e-10 1.059423e-08</span>
<span class="co">## 3  1e-10 1.059423e-08</span>
<span class="co">## 4  1e-10 1.059423e-08</span>
<span class="co">## 5  1e-10 1.059423e-08</span>
<span class="fu">annotateRAV</span><span class="op">(</span><span class="va">PCAmodel</span>, <span class="va">sampleScore2</span><span class="op">)</span>
<span class="co">##                                      Description      NES pvalue      qvalues</span>
<span class="co">## 1       SCHUETZ_BREAST_CANCER_DUCTAL_INVASIVE_UP 2.877257  1e-10 4.847199e-09</span>
<span class="co">## 2     VECCHI_GASTRIC_CANCER_ADVANCED_VS_EARLY_UP 2.826869  1e-10 4.847199e-09</span>
<span class="co">## 3                       LIM_MAMMARY_STEM_CELL_UP 2.708890  1e-10 4.847199e-09</span>
<span class="co">## 4 ANASTASSIOU_MULTICANCER_INVASIVENESS_SIGNATURE 2.688898  1e-10 4.847199e-09</span>
<span class="co">## 5                           BOQUEST_STEM_CELL_UP 2.640382  1e-10 4.847199e-09</span></pre></div>
</div>
</div>
<div id="identified-using-metadata-cms" class="section level2">
<h2 class="hasAnchor">
<a href="#identified-using-metadata-cms" class="anchor"></a>Identified using metadata, CMS</h2>
<p>We find the metadata-associated RAVs using Kruskal-Wallis Rank Sum Test.</p>
<div class="sourceCode" id="cb16"><pre class="downlit">
<span class="va">sampleScore1</span> <span class="op">&lt;-</span> <span class="fl">834</span>
<span class="va">sampleScore2</span> <span class="op">&lt;-</span> <span class="fl">833</span>
<span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span><span class="op">(</span><span class="st">"R/Fig4A_plotting.R"</span>, print.eval <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></pre></div>
<p><img src="repeat_Fig4A_files/figure-html/unnamed-chunk-19-1.png" width="80%" height="80%"></p>
<pre><code>## `summarise()` regrouping output by 'study' (override with `.groups` argument)</code></pre>
<p><img src="repeat_Fig4A_files/figure-html/save_scatter_834_833-1.png" width="700"></p>
<pre><code>## $value
## 
## $visible
## [1] FALSE
## [1] FALSE
## agg_png 
##       2</code></pre>
<div id="wordcloud-2" class="section level3">
<h3 class="hasAnchor">
<a href="#wordcloud-2" class="anchor"></a>wordcloud</h3>
<div class="sourceCode" id="cb19"><pre class="downlit">
<span class="fu">drawWordcloud</span><span class="op">(</span><span class="va">PCAmodel</span>, <span class="va">sampleScore1</span><span class="op">)</span>
<span class="co">## Warning in wordcloud::wordcloud(words = all$word, freq = all$freq, scale =</span>
<span class="co">## scale, : Histocompatibility Testing could not be fit on page. It will not be</span>
<span class="co">## plotted.</span>
<span class="co">## Warning in wordcloud::wordcloud(words = all$word, freq = all$freq, scale =</span>
<span class="co">## scale, : Massively Parallel Sequencing could not be fit on page. It will not be</span>
<span class="co">## plotted.</span>
<span class="co">## Warning in wordcloud::wordcloud(words = all$word, freq = all$freq, scale =</span>
<span class="co">## scale, : Histocompatibility Antigens Class II could not be fit on page. It will</span>
<span class="co">## not be plotted.</span></pre></div>
<p><img src="repeat_Fig4A_files/figure-html/unnamed-chunk-20-1.png" width="60%" height="60%"></p>
<div class="sourceCode" id="cb20"><pre class="downlit">
<span class="fu">drawWordcloud</span><span class="op">(</span><span class="va">PCAmodel</span>, <span class="va">sampleScore2</span><span class="op">)</span>
<span class="co">## Warning in wordcloud::wordcloud(words = all$word, freq = all$freq, scale =</span>
<span class="co">## scale, : Stomach Neoplasms could not be fit on page. It will not be plotted.</span>

<span class="co">## Warning in wordcloud::wordcloud(words = all$word, freq = all$freq, scale =</span>
<span class="co">## scale, : Histocompatibility Antigens Class II could not be fit on page. It will</span>
<span class="co">## not be plotted.</span></pre></div>
<p><img src="repeat_Fig4A_files/figure-html/unnamed-chunk-20-2.png" width="60%" height="60%"></p>
</div>
<div id="gsea-2" class="section level3">
<h3 class="hasAnchor">
<a href="#gsea-2" class="anchor"></a>GSEA</h3>
<div class="sourceCode" id="cb21"><pre class="downlit">
<span class="fu">annotateRAV</span><span class="op">(</span><span class="va">PCAmodel</span>, <span class="va">sampleScore1</span><span class="op">)</span>
<span class="co">##                                             Description      NES pvalue</span>
<span class="co">## 1 ALTEMEIER_RESPONSE_TO_LPS_WITH_MECHANICAL_VENTILATION 2.868397  1e-10</span>
<span class="co">## 2                            MCLACHLAN_DENTAL_CARIES_UP 2.627526  1e-10</span>
<span class="co">## 3                           WIELAND_UP_BY_HBV_INFECTION 2.541545  1e-10</span>
<span class="co">## 4            ZHANG_RESPONSE_TO_IKK_INHIBITOR_AND_TNF_UP 2.339746  1e-10</span>
<span class="co">## 5                       POOLA_INVASIVE_BREAST_CANCER_UP 2.334263  1e-10</span>
<span class="co">##        qvalues</span>
<span class="co">## 1 4.683457e-08</span>
<span class="co">## 2 4.683457e-08</span>
<span class="co">## 3 4.683457e-08</span>
<span class="co">## 4 4.683457e-08</span>
<span class="co">## 5 4.683457e-08</span>
<span class="fu">annotateRAV</span><span class="op">(</span><span class="va">PCAmodel</span>, <span class="va">sampleScore2</span><span class="op">)</span>
<span class="co">##                                  Description      NES pvalue      qvalues</span>
<span class="co">## 1       SENGUPTA_NASOPHARYNGEAL_CARCINOMA_DN 3.158261  1e-10 3.026853e-09</span>
<span class="co">## 2            COLDREN_GEFITINIB_RESISTANCE_DN 2.970178  1e-10 3.026853e-09</span>
<span class="co">## 3                HOLLERN_EMT_BREAST_TUMOR_DN 2.937327  1e-10 3.026853e-09</span>
<span class="co">## 4          WAMUNYOKOLI_OVARIAN_CANCER_LMP_UP 2.931702  1e-10 3.026853e-09</span>
<span class="co">## 5 VECCHI_GASTRIC_CANCER_ADVANCED_VS_EARLY_DN 2.920688  1e-10 3.026853e-09</span></pre></div>
<div class="sourceCode" id="cb22"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">## R version 4.0.3 (2020-10-10)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Ubuntu 18.04.5 LTS</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3</span>
<span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    parallel  stats     graphics  grDevices utils     datasets </span>
<span class="co">## [8] methods   base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] PCAGenomicSignaturesPaper_0.0.1 PCAGenomicSignatures_0.0.5     </span>
<span class="co">##  [3] SummarizedExperiment_1.19.9     GenomicRanges_1.41.6           </span>
<span class="co">##  [5] GenomeInfoDb_1.25.11            IRanges_2.23.10                </span>
<span class="co">##  [7] S4Vectors_0.27.14               MatrixGenerics_1.1.8           </span>
<span class="co">##  [9] matrixStats_0.57.0              forcats_0.5.0                  </span>
<span class="co">## [11] stringr_1.4.0                   dplyr_1.0.2                    </span>
<span class="co">## [13] purrr_0.3.4                     readr_1.4.0                    </span>
<span class="co">## [15] tidyr_1.1.2                     tibble_3.0.4                   </span>
<span class="co">## [17] ggplot2_3.3.2                   tidyverse_1.3.0                </span>
<span class="co">## [19] Biobase_2.49.1                  BiocGenerics_0.35.4            </span>
<span class="co">## [21] BiocStyle_2.17.2               </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##  [1] colorspace_1.4-1       ggsignif_0.6.0         rjson_0.2.20          </span>
<span class="co">##  [4] ellipsis_0.3.1         rio_0.5.16             rprojroot_1.3-2       </span>
<span class="co">##  [7] circlize_0.4.10        XVector_0.29.3         GlobalOptions_0.1.2   </span>
<span class="co">## [10] fs_1.5.0               clue_0.3-57            rstudioapi_0.11       </span>
<span class="co">## [13] farver_2.0.3           ggpubr_0.4.0           bit64_4.0.5           </span>
<span class="co">## [16] fansi_0.4.1            lubridate_1.7.9        xml2_1.3.2            </span>
<span class="co">## [19] knitr_1.30             jsonlite_1.7.1         broom_0.7.2           </span>
<span class="co">## [22] cluster_2.1.0          dbplyr_1.4.4           png_0.1-7             </span>
<span class="co">## [25] BiocManager_1.30.10    compiler_4.0.3         httr_1.4.2            </span>
<span class="co">## [28] backports_1.1.10       assertthat_0.2.1       Matrix_1.2-18         </span>
<span class="co">## [31] cli_2.1.0              htmltools_0.5.0        tools_4.0.3           </span>
<span class="co">## [34] gtable_0.3.0           glue_1.4.2             GenomeInfoDbData_1.2.4</span>
<span class="co">## [37] rappdirs_0.3.1         Rcpp_1.0.5             carData_3.0-4         </span>
<span class="co">## [40] cellranger_1.1.0       pkgdown_1.6.1          vctrs_0.3.4           </span>
<span class="co">## [43] xfun_0.18              openxlsx_4.2.2         rvest_0.3.6           </span>
<span class="co">## [46] lifecycle_0.2.0        rstatix_0.6.0          zlibbioc_1.35.0       </span>
<span class="co">## [49] scales_1.1.1           ragg_0.4.0             hms_0.5.3             </span>
<span class="co">## [52] RColorBrewer_1.1-2     ComplexHeatmap_2.5.6   yaml_2.2.1            </span>
<span class="co">## [55] curl_4.3               memoise_1.1.0          RSQLite_2.2.1         </span>
<span class="co">## [58] stringi_1.5.3          desc_1.2.0             zip_2.1.1             </span>
<span class="co">## [61] shape_1.4.5            rlang_0.4.8            pkgconfig_2.0.3       </span>
<span class="co">## [64] systemfonts_0.3.2      bitops_1.0-6           evaluate_0.14         </span>
<span class="co">## [67] lattice_0.20-41        labeling_0.4.2         bit_4.0.4             </span>
<span class="co">## [70] tidyselect_1.1.0       magrittr_1.5           bookdown_0.21         </span>
<span class="co">## [73] R6_2.4.1               generics_0.0.2         DelayedArray_0.15.16  </span>
<span class="co">## [76] DBI_1.1.0              pillar_1.4.6           haven_2.3.1           </span>
<span class="co">## [79] foreign_0.8-79         withr_2.3.0            abind_1.4-5           </span>
<span class="co">## [82] RCurl_1.98-1.2         modelr_0.1.8           crayon_1.3.4          </span>
<span class="co">## [85] car_3.0-10             wordcloud_2.6          BiocFileCache_1.13.1  </span>
<span class="co">## [88] rmarkdown_2.5          GetoptLong_1.0.4       grid_4.0.3            </span>
<span class="co">## [91] readxl_1.3.1           data.table_1.13.2      blob_1.2.1            </span>
<span class="co">## [94] reprex_0.3.0           digest_0.6.27          textshaping_0.1.2     </span>
<span class="co">## [97] munsell_0.5.0</span></pre></div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Sehyun Oh.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
