<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Training datasets for RAVmodel building • GenomicSuperSignaturePaper</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Training datasets for RAVmodel building">
<meta property="og:description" content="GenomicSuperSignaturePaper">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">GenomicSuperSignaturePaper</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RAVmodel
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/training_datasets/available_samples.html">Training Data</a>
    </li>
    <li>
      <a href="../../articles/select_numOfClusters/find_optimum_k.html">Number of clusters</a>
    </li>
    <li>
      <a href="../../articles/others/repeat_Fig5_Pathway_Separation.html">Pathway separation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    EDA
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/TCGA/TCGA_5studies.html">Database search</a>
    </li>
    <li>
      <a href="../../articles/E-MTAB-2452/E-MTAB-2452.html">Annoated PCA</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Subtyping
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/CRC/CRC_Subtyping.html">CRC subtype scores</a>
    </li>
    <li>
      <a href="../../articles/CRC/PCSS_vs_RAV.html">RAVs similar to PCSS1/2</a>
    </li>
    <li>
      <a href="../../articles/CRC/metadata_based_search.html">Metadata based search</a>
    </li>
    <li>
      <a href="../../articles/CRC/CRC_ClinicalVar.html">CRC clinical variables</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Transfer Learning
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/SLE-WB/neutrophil_counts_SLE-WB.html">Neutrophil (SLE-WB)</a>
    </li>
    <li>
      <a href="../../articles/NARES/neutrophil_counts_NARES.html">Neutrophil (NARES)</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/shbrief/GenomicSuperSignaturePaper">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="available_samples_files/header-attrs-2.9/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Training datasets for RAVmodel building</h1>
                        <h4 class="author">Sehyun Oh</h4>
            
            <h4 class="date">2021-07-14</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/shbrief/GenomicSuperSignaturePaper/blob/master/vignettes/training_datasets/available_samples.Rmd"><code>vignettes/training_datasets/available_samples.Rmd</code></a></small>
      <div class="hidden name"><code>available_samples.Rmd</code></div>

    </div>

    
    
<div id="source" class="section level1">
<h1 class="hasAnchor">
<a href="#source" class="anchor"></a>Source</h1>
<p>We used human RNA sequencing datasets from RNA-seq Sample Compendia in <a href="https://www.refine.bio/">refine.bio</a>, which hosts uniformly processed gene expression data from EBI’s ArrayExpress, NCBI’s GEO and SRA. Data were downloaded on April 10th, 2020, and we selected studies based on the following criteria: 1) Exclude studies with more than 1,000 samples because they are more likely to be single-cell RNA sequencing datasets. 2) Exclude studies assigned with a Medical Subject Headings (MeSH) term, “Single-Cell Analysis”. 3) Exclude studies with less than or equal to 50 successfully downloaded and imported samples. After filtering, the complete compendium includes 536 studies (defined as a single SRA study) comprising 44,890 samples.</p>
<p>The detail of training data is summarized in <code>studyMeta.tsv</code> table in <code>GenomicSuperSignature</code> package. This vignette explains how this summary table is built.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">extdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"GenomicSuperSignature"</span><span class="op">)</span>
<span class="va">studyMeta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">extdata</span>, <span class="st">"studyMeta.tsv.gz"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">studyMeta</span><span class="op">)</span>
<span class="co">##   studyName downloaded metadata imported RAVmodel_677 RAVmodel_1399</span>
<span class="co">## 2 DRP000425          4        4    FALSE        FALSE         FALSE</span>
<span class="co">## 3 DRP000464          8        8    FALSE        FALSE         FALSE</span>
<span class="co">## 4 DRP000499         21       21     TRUE        FALSE          TRUE</span>
<span class="co">## 5 DRP000527          2        4    FALSE        FALSE         FALSE</span>
<span class="co">## 6 DRP000665          7        7    FALSE        FALSE         FALSE</span>
<span class="co">## 7 DRP000929          4        4    FALSE        FALSE         FALSE</span>
<span class="co">##   RAVmodel_536</span>
<span class="co">## 2        FALSE</span>
<span class="co">## 3        FALSE</span>
<span class="co">## 4        FALSE</span>
<span class="co">## 5        FALSE</span>
<span class="co">## 6        FALSE</span>
<span class="co">## 7        FALSE</span>
<span class="co">##                                                                                                                                                                     title</span>
<span class="co">## 2                                                                                                           Homo sapiens T24 human bladder cancer cell line Transcriptome</span>
<span class="co">## 3 pre-miRNA profiles obtained through application of locked nucleic acids reveals complex 5’/3’ arm variation including concomitant cleavage and polyuridylation patterns</span>
<span class="co">## 4                                                                                                                      Homo sapiens strain:Human ICESeq Genome sequencing</span>
<span class="co">## 5                                                                                                      RNA sequencing of wild-type or mutant U2AF35 transduced HeLa Cells</span>
<span class="co">## 6                                                                                                             Global transcriptional response against glucose deprivation</span>
<span class="co">## 7                                                                  Human inactive X chromosome is compacted through a polycomb-independent SMCHD1-HBiX1 pathway [RNA-seq]</span></code></pre></div>
</div>
<div id="availability-metadata" class="section level1">
<h1 class="hasAnchor">
<a href="#availability-metadata" class="anchor"></a>Availability: metadata</h1>
<p><code>aggregated_metadata.json</code> is downloaded along with the training dataset. We collect the number of samples per study based on this metadata.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data.dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"GenomicSuperSignaturePaper"</span><span class="op">)</span>
<span class="va">meta_json</span> <span class="op">&lt;-</span> <span class="fu">jsonlite</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/jsonlite/man/fromJSON.html">fromJSON</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data.dir</span>, 
                                          <span class="st">"aggregated_metadata.json"</span><span class="op">)</span><span class="op">)</span>
<span class="va">experiments</span> <span class="op">&lt;-</span> <span class="va">meta_json</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span> 

<span class="va">studies</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_along</a></span><span class="op">(</span><span class="va">experiments</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">studies</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/append.html">append</a></span><span class="op">(</span><span class="va">studies</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">experiments</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">sample_accession_codes</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">studies</span><span class="op">)</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">experiments</span><span class="op">)</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>
<span class="op">}</span>

<span class="va">sampleNum</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">studies</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="availability-downloadimport" class="section level1">
<h1 class="hasAnchor">
<a href="#availability-downloadimport" class="anchor"></a>Availability: download/import</h1>
<p>I summarized the number of samples based on <code>aggregated_metadata.json</code> under <code>metadata</code> column and the number of actually downloaded <code>_quant.sf</code> sample files under <code>downloaded</code> column in the <code>studyMeta</code> table.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># download.dir &lt;- "path/to/download/directory"</span>
<span class="va">dirNames</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">download.dir</span><span class="op">)</span>

<span class="va">ind_rm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">dirNames</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> 
                  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"aggregated_metadata.json"</span>, <span class="st">"README.md"</span>, <span class="st">"LICENSE.TXT"</span><span class="op">)</span><span class="op">)</span>
<span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">ind_rm</span><span class="op">)</span> <span class="op">!=</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span><span class="va">dirNames</span> <span class="op">&lt;-</span> <span class="va">dirNames</span><span class="op">[</span><span class="op">-</span><span class="va">ind_rm</span><span class="op">]</span><span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">dirNames</span><span class="op">)</span> <span class="co"># 6457 studies were downloaded</span>

<span class="va">studyMeta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span><span class="cn">NA</span>, nrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">dirNames</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="fl">3</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">studyMeta</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"studyName"</span>, <span class="st">"downloaded"</span>, <span class="st">"metadata"</span><span class="op">)</span> 
<span class="va">studyMeta</span><span class="op">$</span><span class="va">studyName</span> <span class="op">&lt;-</span> <span class="va">dirNames</span>

<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_along</a></span><span class="op">(</span><span class="va">dirNames</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">dir_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">download.dir</span>, <span class="va">dirNames</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span>
  
  <span class="co">## actually downloaded samples</span>
  <span class="va">ind1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">studyMeta</span><span class="op">$</span><span class="va">studyName</span> <span class="op">==</span> <span class="va">dirNames</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span> 
  <span class="va">studyMeta</span><span class="op">$</span><span class="va">downloaded</span><span class="op">[</span><span class="va">ind1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"_quant.sf"</span>, 
                                     <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">dir_path</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="va">length</span>
   
  <span class="co">## available samples based on metadata</span>
  <span class="va">ind2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">sampleNum</span><span class="op">)</span> <span class="op">==</span> <span class="va">dirNames</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span>  
  <span class="va">studyMeta</span><span class="op">$</span><span class="va">metadata</span><span class="op">[</span><span class="va">ind2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">sampleNum</span><span class="op">[</span><span class="va">ind2</span><span class="op">]</span>
<span class="op">}</span></code></pre></div>
</div>
<div id="imported-samples" class="section level1">
<h1 class="hasAnchor">
<a href="#imported-samples" class="anchor"></a>Imported samples</h1>
<p>1399 studies with more than 20 samples were successfully imported by <code><a href="https://rdrr.io/pkg/tximport/man/tximport.html">tximport::tximport</a></code></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">studyNames</span> <span class="op">&lt;-</span> <span class="va">studyMeta</span><span class="op">$</span><span class="va">studyName</span>
<span class="va">importedFiles</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">)</span>

<span class="co"># download.dir2 &lt;- "path/to/imported/directory"</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_along</a></span><span class="op">(</span><span class="va">studyNames</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">out.path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">download.dir2</span>, <span class="va">studyNames</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span>
  <span class="va">fname</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">out.path</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">studyNames</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>, <span class="st">".rds"</span><span class="op">)</span><span class="op">)</span>
  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span><span class="op">(</span><span class="va">fname</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">importedFiles</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">importedFiles</span>, <span class="va">studyNames</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span>
  <span class="op">}</span>
<span class="op">}</span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">imported_ind</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">studyMeta</span><span class="op">$</span><span class="va">studyName</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">importedFiles</span><span class="op">)</span>
<span class="va">studyMeta</span><span class="op">$</span><span class="va">imported</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span>
<span class="va">studyMeta</span><span class="op">$</span><span class="va">imported</span><span class="op">[</span><span class="va">imported_ind</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span> </code></pre></div>
<p>Study title is also added.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_along</a></span><span class="op">(</span><span class="va">experiments</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">title</span> <span class="op">&lt;-</span> <span class="va">experiments</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">title</span>
  <span class="va">studyMeta</span><span class="op">$</span><span class="va">title</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">title</span>
<span class="op">}</span></code></pre></div>
</div>
<div id="summary-plot" class="section level1">
<h1 class="hasAnchor">
<a href="#summary-plot" class="anchor"></a>Summary plot</h1>
<p>Metadata bar (light blue) shows the number of studies with the given ranges of study sizes based on the metadata. Downloaded bar (pink) represents the number of studies with the given ranges of study sizes that were successfully downloaded and imported through <code>tximport</code>. Based on metadata, there were studies with more than 100 samples, but at the time of snapshot, only up to 100 samples were available. Thus, the plot displays only up to 100 samples/study cases. Due to the unavailability of certain samples, more studies belong to 0-5 samples/study bracket than metadata suggests.</p>
<p><img src="available_samples_files/figure-html/unnamed-chunk-10-1.png" width="576"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Cases where all the available samples (based on metadata) were downloaded</span>
<span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">studyMeta</span><span class="op">$</span><span class="va">downloaded</span> <span class="op">==</span> <span class="va">studyMeta</span><span class="op">$</span><span class="va">metadata</span><span class="op">)</span>
<span class="co">## [1] 3135</span>

<span class="co">## The number of studies with more than 100 samples: </span>
<span class="co">## Actually downloaded vs. expected by metadata</span>
<span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">studyMeta</span><span class="op">$</span><span class="va">downloaded</span> <span class="op">&gt;</span> <span class="fl">100</span><span class="op">)</span>
<span class="co">## [1] 0</span>
<span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">studyMeta</span><span class="op">$</span><span class="va">metadata</span> <span class="op">&gt;</span> <span class="fl">100</span><span class="op">)</span>
<span class="co">## [1] 426</span></code></pre></div>
<div id="example" class="section level3">
<h3 class="hasAnchor">
<a href="#example" class="anchor"></a>Example</h3>
<p>Example study, <code>SRP152576</code>:<br>
- 13,127 samples are available based on <code>aggregated_metadata.json</code>.<br>
- No sample is available based on <a href="https://www.refine.bio/">refine.bio</a> webpage.<br>
- 100 samples are actually downloaded.<br>
- 3,927 samples are available based on <code>metadata_SRP152576.csv</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ind</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.min.html">which.max</a></span><span class="op">(</span><span class="va">studyMeta</span><span class="op">$</span><span class="va">metadata</span><span class="op">)</span>  
<span class="va">studyMeta</span><span class="op">[</span><span class="va">ind</span>,<span class="op">]</span>
<span class="co">##      studyName downloaded metadata imported RAVmodel_677 RAVmodel_1399</span>
<span class="co">## 5531 SRP152576        100    13127     TRUE        FALSE          TRUE</span>
<span class="co">##      RAVmodel_536</span>
<span class="co">## 5531        FALSE</span>
<span class="co">##                                                                                title</span>
<span class="co">## 5531 In vivo molecular signatures of severe dengue infection revealed by viscRNA-Seq</span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">file_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">download.dir</span>, <span class="va">studyMeta</span><span class="op">$</span><span class="va">studyName</span><span class="op">[</span><span class="va">ind</span><span class="op">]</span><span class="op">)</span>
<span class="va">fname</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"metadata_"</span>, <span class="va">studyMeta</span><span class="op">$</span><span class="va">studyName</span><span class="op">[</span><span class="va">ind</span><span class="op">]</span>, <span class="st">".tsv"</span><span class="op">)</span>
<span class="va">sampleMeta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">file_dir</span>, <span class="va">fname</span><span class="op">)</span>, header <span class="op">=</span> <span class="cn">TRUE</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">sampleMeta</span><span class="op">)</span>
<span class="co">## [1] 3927   19</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">sampleMeta</span>, <span class="fl">3</span><span class="op">)</span>
<span class="co">##   refinebio_accession_code experiment_accession refinebio_age</span>
<span class="co">## 1               SRR7481117            SRP152576            14</span>
<span class="co">## 2               SRR7480712            SRP152576            14</span>
<span class="co">## 3               SRR7483223            SRP152576            31</span>
<span class="co">##   refinebio_cell_line refinebio_compound refinebio_disease</span>
<span class="co">## 1                  NA                 NA                NA</span>
<span class="co">## 2                  NA                 NA                NA</span>
<span class="co">## 3                  NA                 NA                NA</span>
<span class="co">##   refinebio_disease_stage refinebio_genetic_information refinebio_organism</span>
<span class="co">## 1                      NA                            NA       HOMO_SAPIENS</span>
<span class="co">## 2                      NA                            NA       HOMO_SAPIENS</span>
<span class="co">## 3                      NA                            NA       HOMO_SAPIENS</span>
<span class="co">##         refinebio_platform refinebio_race refinebio_sex</span>
<span class="co">## 1 NextSeq 500 (NextSeq500)             NA          male</span>
<span class="co">## 2 NextSeq 500 (NextSeq500)             NA          male</span>
<span class="co">## 3 NextSeq 500 (NextSeq500)             NA        female</span>
<span class="co">##   refinebio_source_archive_url refinebio_source_database</span>
<span class="co">## 1                           NA                       SRA</span>
<span class="co">## 2                           NA                       SRA</span>
<span class="co">## 3                           NA                       SRA</span>
<span class="co">##   refinebio_specimen_part refinebio_subject refinebio_time refinebio_title</span>
<span class="co">## 1                   blood     blood (pbmcs)             NA  1001701103_E10</span>
<span class="co">## 2                   blood     blood (pbmcs)             NA  1001701101_C19</span>
<span class="co">## 3                   blood     blood (pbmcs)             NA  1001701401_P23</span>
<span class="co">##   refinebio_treatment</span>
<span class="co">## 1                  NA</span>
<span class="co">## 2                  NA</span>
<span class="co">## 3                  NA</span></code></pre></div>
</div>
</div>
<div id="studies-used-for-model-buildling" class="section level1">
<h1 class="hasAnchor">
<a href="#studies-used-for-model-buildling" class="anchor"></a>Studies used for model buildling</h1>
<div id="ravmodel_677" class="section level2">
<h2 class="hasAnchor">
<a href="#ravmodel_677" class="anchor"></a>RAVmodel_677</h2>
<p><code>studies_with_error.csv</code> file contains the sample names that were failed to be downloaded/imported when I tried to use studies &gt; 50 samples based on the metadata. You can find more documentation on this under <code>Methods/model_building</code>.</p>
</div>
<div id="ravmodel_1399" class="section level2">
<h2 class="hasAnchor">
<a href="#ravmodel_1399" class="anchor"></a>RAVmodel_1399</h2>
<p>All the studies that have more than 20 successfully imported samples. So, it’s practically same as <code>imported</code> column.</p>
</div>
<div id="ravmodel_536" class="section level2">
<h2 class="hasAnchor">
<a href="#ravmodel_536" class="anchor"></a>RAVmodel_536</h2>
<p>These 536 studies meet the criteria, &gt; 50 (downloaded) and &lt;1000 (metadata) samples per study. Also, any study with “Single-Cell Analysis” MeSH term was removed.</p>
</div>
</div>
<div id="multiplier-vs--ravmodel" class="section level1">
<h1 class="hasAnchor">
<a href="#multiplier-vs--ravmodel" class="anchor"></a>MultiPLIER vs. RAVmodel</h1>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Training data used for GenomicSuperSignature</span>
<span class="va">res_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="st">"accessions.rds"</span><span class="op">)</span>

<span class="co">## Training data used for multiPLIER</span>
<span class="va">recount2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="st">"https://raw.githubusercontent.com/greenelab/multi-plier/master/data/sample_info/recount2_srr_srs_srp.tsv"</span>, header <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">res_all</span>, <span class="fl">3</span><span class="op">)</span>
<span class="co">##   run_accession source study_accession sample_accession experiment_accession</span>
<span class="co">## 1     DRR006374    SRA       DRP000987        DRS005720            DRX005592</span>
<span class="co">## 2     DRR006375    SRA       DRP000987        DRS005721            DRX005593</span>
<span class="co">## 3     DRR006376    SRA       DRP000987        DRS005722            DRX005594</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">recount2</span>, <span class="fl">3</span><span class="op">)</span>
<span class="co">##         run     study    sample</span>
<span class="co">## 1 SRR013549 SRP000599 SRS002125</span>
<span class="co">## 2 SRR013550 SRP000599 SRS002126</span>
<span class="co">## 3 SRR013551 SRP000599 SRS002127</span></code></pre></div>
<ol style="list-style-type: decimal">
<li>multiPLIER used 37027 unique runs from 30301 samples in 1466 studies.<br>
</li>
<li>GenomicSuperSignature used 44890 unique runs from 34616 samples in 535 studies.<br>
</li>
<li>There are 6839 overlapping runs from 5260 samples in 87 studies used for both models.</li>
</ol>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<details><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</details>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Sehyun Oh.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
