<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Reproduce GenomicSuperSignature paper • GenomicSuperSignaturePaper</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Reproduce GenomicSuperSignature paper">
<meta property="og:description" content="This package provides data, methods, results covered in GenomicSuperSignature manuscript. ">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">GenomicSuperSignaturePaper</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RAVmodel
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/training_datasets/available_samples.html">Training Data</a>
    </li>
    <li>
      <a href="articles/select_numOfClusters/find_optimum_k.html">Number of clusters</a>
    </li>
    <li>
      <a href="articles/others/repeat_Fig5_Pathway_Separation.html">Pathway separation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    EDA
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/TCGA/TCGA_5studies.html">Database search</a>
    </li>
    <li>
      <a href="articles/E-MTAB-2452/E-MTAB-2452.html">Annoated PCA</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Subtyping
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/CRC/CRC_Subtyping.html">CRC subtype scores</a>
    </li>
    <li>
      <a href="articles/CRC/PCSS_vs_RAV.html">RAVs similar to PCSS1/2</a>
    </li>
    <li>
      <a href="articles/CRC/metadata_based_search.html">Metadata based search</a>
    </li>
    <li>
      <a href="articles/CRC/CRC_ClinicalVar.html">CRC clinical variables</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Transfer Learning
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/SLE-WB/neutrophil_counts_SLE-WB.html">Neutrophil (SLE-WB)</a>
    </li>
    <li>
      <a href="articles/NARES/neutrophil_counts_NARES.html">Neutrophil (NARES)</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/shbrief/GenomicSuperSignaturePaper">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="genomicsupersignaturepaper" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#genomicsupersignaturepaper" class="anchor"></a>GenomicSuperSignaturePaper</h1></div>
<p>Here, we presents the usecases and benchmarking examples of <a href="http://www.bioconductor.org/packages/release/bioc/html/GenomicSuperSignature.html">GenomicSuperSignature</a> R/Bioconductor package. More description on these examples can be found in the <a href="https://www.biorxiv.org/content/10.1101/2021.05.26.445900v1">manuscript</a>.</p>
<div id="disease-subtyping" class="section level2">
<h2 class="hasAnchor">
<a href="#disease-subtyping" class="anchor"></a>Disease Subtyping</h2>
<p>To compare the utility of GenomicSuperSignature relative to the focused use of data from a single disease, we compared RAVs to <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-018-1511-4">the previous study</a> (referred as <em>CRC paper</em> below) that employed colorectal cancer (CRC) gene expression databases to identify CRC molecular subtypes.</p>
<div id="continuous-subtype-score" class="section level4">
<h4 class="hasAnchor">
<a href="#continuous-subtype-score" class="anchor"></a>Continuous Subtype Score</h4>
<p>The left panel is Figure 4A from CRC paper and the right panel is the CRC subtype separation by GenomicSuperSignature. CRC paper built the continuous score model based on the 8 colorectal cancer specific, microarray datasets with graph-based clustering method. GenomicSuperSignature was generated from 536 heterogeneous, RNA sequencing datasets and used hierarchical clustering. Despite of these major differences in model building process, GenomicSuperSignature and CRC paper’s model show comparable performance.</p>
<p><img src="https://raw.githubusercontent.com/shbrief/GenomicSuperSignaturePaper/master/inst/images/CRC_Fig4A.png" width="300" height="300"><img src="https://raw.githubusercontent.com/shbrief/GenomicSuperSignaturePaper/master/Results/CRC/outputs/png/scatter_834_833.png" width="300" height="300"></p>
<p>Details on these plots are described in <a href="https://shbrief.github.io/GenomicSuperSignaturePaper/articles/CRC/CRC_Subtyping.html">Subtyping/CRC subtype scores</a>.</p>
</div>
<div id="clinical-variables" class="section level4">
<h4 class="hasAnchor">
<a href="#clinical-variables" class="anchor"></a>Clinical Variables</h4>
<p>Below is Figure 4 from CRC paper. LRT (likelihood-ratio test) between the reference full model versus either the continuous scores only model or the discrete subtypes only model are plotted here. A p-value near 1 (−log10p-value near 0) suggests that no additional information is provided by the full model.</p>
<p>This results shows that PCSS1/2 (the second column, <em>score</em>) provide additional information in characterizing molecular/histological/clinical correlates than CMS (the first column, <em>subtype</em>).</p>
<p><img src="https://raw.githubusercontent.com/shbrief/GenomicSuperSignaturePaper/master/inst/images/CRC_Fig4C.png" width="500" height="300"></p>
<p></p>
<p>RAV834/833, two RAVs that explain CMS subtypes best among 4,764 RAVs, also performs better than CMS subtypes.</p>
<p><img src="https://raw.githubusercontent.com/shbrief/GenomicSuperSignaturePaper/master/Results/CRC/outputs/png/boxplot_CMS_vs_834_833.png" width="500" height="300"></p>
<p></p>
<p>We directly compared the performance between PCSSs and RAVs. Except the tumor location (the last panel), RAV834/833-based model out-performed PCSS1/2-based model relative to the combined model.</p>
<p><img src="https://raw.githubusercontent.com/shbrief/GenomicSuperSignaturePaper/master/Results/CRC/outputs/png/boxplot_PCSS_vs_834_833.png" width="500" height="300"></p>
<p></p>
<p>Details on the above plots are described in <a href="https://shbrief.github.io/GenomicSuperSignaturePaper/articles/CRC/CRC_ClinicalVar.html">Subtyping/CRC clinical variables</a>.</p>
</div>
</div>
<div id="transfer-learning" class="section level2">
<h2 class="hasAnchor">
<a href="#transfer-learning" class="anchor"></a>Transfer Learning</h2>
<p>For practical and technical reasons, biological datasets often contain missing information or signals buried in noise. GenomicSuperSignature can fill out those gaps by uncovering weak or indirectly measured biological attributes of a new dataset by leveraging the existing databases. To evaluate this ‘transfer learning’ aspect of the GenomicSuperSignature, we compared the neutrophil count estimation by RAVs across two different datasets, systemic lupus erythematosus whole blood (<a href="https://www.cell.com/cell/pdfExtended/S0092-8674(16)30264-1">SLE-WB</a>) and nasal brushing (<a href="https://onlinelibrary.wiley.com/doi/full/10.1002/art.39185">NARES</a>) datasets as described in <a href="https://www.cell.com/cell-systems/fulltext/S2405-4712(19)30119-X?_returnURL=https%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS240547121930119X%3Fshowall%3Dtrue">the previous study</a>(referred as <em>MultiPLIER paper</em> below).</p>
<div id="neutrophil-countsestimates" class="section level4">
<h4 class="hasAnchor">
<a href="#neutrophil-countsestimates" class="anchor"></a>Neutrophil Counts/Estimates</h4>
<p>The top panel is Figure 3 from MultiPLIER paper and the bottom panel is the result from GenomicSuperSignature. In MultiPLIER paper, latent variables (LVs) were built from recount2 datasets using <a href="https://www.nature.com/articles/s41592-019-0456-1">Pathway-level information extractor (PLIER)</a> method. Despite of the major differences in model building process, GenomicSuperSignature and MultiPLIER paper’s results show comparable performance.</p>
<p>Neutrophil-associated LV, LV603, from MultiPLIER model shows the positive correlation with neutrophil counts/estimates of two different datasets, SLE-WB and NARES.</p>
<p><img src="https://raw.githubusercontent.com/shbrief/GenomicSuperSignaturePaper/master/inst/images/multiPLIER_Fig3.png" width="600" height="200"></p>
<p> We used SLE-WB dataset to identify the neutrophil-associated RAV, RAV1551, and applied it to NARES dataset.</p>
<p><img src="https://raw.githubusercontent.com/shbrief/GenomicSuperSignaturePaper/master/Results/multiPLIER_Fig3.png" width="600" height="200"></p>
<p>Details on the analysis process are described in <a href="https://shbrief.github.io/GenomicSuperSignaturePaper/articles/SLE-WB/neutrophil_counts_SLE-WB.html">Transfer Learning/Neutrophil (SLE-WB)</a> and <a href="https://shbrief.github.io/GenomicSuperSignaturePaper/articles/NARES/neutrophil_counts_NARES.html">Transfer Learning/Neutrophil (NARES)</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/shbrief/GenomicSuperSignaturePaper/">https://​github.com/​shbrief/​GenomicSuperSignaturePaper/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/shbrief/GenomicSuperSignaturePaper/issues">https://​github.com/​shbrief/​GenomicSuperSignaturePaper/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Sehyun Oh <br><small class="roles"> Author, maintainer </small>  </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Sehyun Oh.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
