---
title: "PCA on E-MTAB-2452"
author: "Sehyun Oh"
date: "`r format(Sys.time(), '%B %d, %Y')`"
output:
  BiocStyle::html_document:
    toc: true
    toc_float: false
    toc_depth: 3
vignette: >
  %\VignetteIndexEntry{PCA E-MTAB-2452}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
abstract: "[**Source Code**](https://github.com/shbrief/GenomicSuperSignaturePaper/blob/master/Results/E-MTAB-2452/E-MTAB-2452.Rmd)"
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  comment = "#>", message = FALSE, warning = TRUE, collapse = TRUE,
  out.height="75%", out.width="75%"
)
```


# Setup
One of the widely used exploratory data analysis methods is PCA and a PCA plot can provide a quick overview of sample composition and distribution. However, the interpretation of different PCs is not readily available in the conventional PCA. We couple PCs from new data with GSEA annotation of RAVmodel and enable the instant interpretation of PCA results. Here, we show this example using a microarray dataset from isolated immune cells (E-MTAB-2452) and RAVmodel annotated with three priors from the PLIER package (RAVmodel_PLIERpriors).

## Load packages
```{r}
suppressPackageStartupMessages({
  library(dplyr)
  library(GenomicSuperSignature)
  library(EBImage)
})
```

## E-MTAB-2452
[E-MTAB-2452](https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-2452/) 
contains sorted peripheral blood cells (CD4+ T cells, CD14+ monocytes, 
CD16+ neutrophils) profiled on microarray from several autoimmune diseases.   

The expression data of E-MTAB-2452 is pre-processed by Greene lab and available [here](https://github.com/greenelab/multi-plier/blob/master/data/expression_data/E-MTAB-2452_hugene11st_SCANfast_with_GeneSymbol.pcl).

```{r}
annot.dat <- readr::read_tsv("data/E-MTAB-2452_hugene11st_SCANfast_with_GeneSymbol.pcl") %>% as.data.frame
rownames(annot.dat) <- annot.dat[, 2]

dataset <- as.matrix(annot.dat[, 3:ncol(annot.dat)])
rownames(dataset) <- annot.dat$GeneSymbol
dataset[1:3, 1:3]
```

Each sample is labeled with the known cell type.
```{r}
cellType <- gsub("_.*$", "", colnames(dataset))
cellType <- gsub("CD4", "CD4,T cell", cellType)
cellType <- gsub("CD14", "CD14,monocyte", cellType)
cellType <- gsub("CD16", "CD16,neutrophil", cellType)
names(cellType) <- colnames(dataset)
```

## RAVmodel
Because E-MTAB-2452 data is comprised of isolated immune subsets and was 
analyzed in MultiPLIER paper with three priors implemented in PLIER method 
by default, we used the RAVmodel annotated with the same PLIER priors.

```{r message=FALSE, warning=FALSE, echo=FALSE}
## If GenomicSuperSignaturePaper is built locally with RAVmodel in inst/extdata
data.dir <- system.file("extdata", package = "GenomicSuperSignaturePaper")
RAVmodel <- readRDS(file.path(data.dir, "RAVmodel_PLIERpriors.rds"))
```

```{r eval=FALSE}
RAVmodel <- getModel("PLIERpriors", load=TRUE)
```

```{r}
RAVmodel

version(RAVmodel)
```

# Annotate top PCs
We checked the enriched pathways for top eight PCs of E-MTAB-2452 data. By 
default, RAVs with the validation score above 0.5 are returned from 
`annotatePC` function. Here, only PC1 and PC2 have the associated RAV under 
the default condition.

```{r}
val_all <- validate(dataset, RAVmodel)
annotatePC(1:8, val_all, RAVmodel, n = 5, simplify = TRUE)  
```

We lowered the validation score cutoff to 0 (`scoreCutoff = 0`) and the output 
returned with more enriched pathways. Top four PCs of E-MTAB-2452 data are 
associated with RAV23/ RAV1552/ RAV1387/ RAV2766.

```{r}
annotatePC(1:8, val_all, RAVmodel, n = 5, simplify = TRUE, scoreCutoff = 0)
```



# Annotated PCA plot
## Pairs plot
To compare which PC separates the different cell types the best, we draw the 
pairs plot with the top four PCs. 

```{r echo=FALSE, fig.width=4.8, fig.height=5.5}
## Data cleaning
library(tibble)
res <- stats::prcomp(dataset)$rotation %>% as.data.frame
res <- tibble::rownames_to_column(res, "SampleID")
ct <- data.frame(cellType)
ct <- tibble::rownames_to_column(ct, "SampleID")

res.df <- dplyr::full_join(res, ct, by = "SampleID") %>%
  dplyr::mutate(PC1 = as.numeric(as.character(PC1)),
                PC2 = as.numeric(as.character(PC2)),
                PC3 = as.numeric(as.character(PC3)),
                PC4 = as.numeric(as.character(PC4)),
                Dataset = factor(cellType, levels = c("CD14,monocyte", 
                                                      "CD16,neutrophil", 
                                                      "CD4,T cell")))

## Pairs plot
PC1 <- res.df[,"PC1"]
PC2 <- res.df[,"PC2"]
PC3 <- res.df[,"PC3"]
PC4 <- res.df[,"PC4"]
data <- data.frame(PC1, PC2, PC3, PC4)
group <- as.factor(res.df$cellType)

pairs(data, col = group, oma=c(10,3,3,3), pch = c(19), cex = 1)
par(xpd = TRUE)
legend("bottom", fill = unique(group), legend = c(levels(group)), cex = 0.7)
```

</br>
Tried `GGally::ggpairs`...

```{r echo=FALSE, fig.width=6.5, fig.height=5}
library(GGally)
ggpairs(data, aes(color = group), columns = 1:4, upper = "blank",
        legend = c(1,1))
```

</br>
Top three PCs seem to separate different cell types well. So we draw 
annotated PCA plot with the different pairs of the top three PCs below. 

## PC1 vs. PC2
```{r fig.width=7, fig.height=7}
plotAnnotatedPCA(dataset, RAVmodel, PCnum = c(1,2), val_all = val_all, 
                 scoreCutoff = 0.3, color_by = cellType, 
                 color_lab = "Cell Type", trimed_pathway_len = 45)
```

## PC1 vs. PC3
```{r fig.width=7, fig.height=7}
plotAnnotatedPCA(dataset, RAVmodel, PCnum = c(1,3), val_all = val_all, 
                 scoreCutoff = 0.3, color_by = cellType, 
                 color_lab = "Cell Type", trimed_pathway_len = 45)
```

## PC2 vs. PC3
```{r fig.width=7, fig.height=7}
plotAnnotatedPCA(dataset, RAVmodel, PCnum = c(2,3), val_all = val_all, 
                 scoreCutoff = 0.3, color_by = cellType, 
                 color_lab = "Cell Type", trimed_pathway_len = 45)
```


# Explore knowledge graph
## Validated RAVs
Among the RAVs associated with the top four PCs of E-MTAB-2452 data, two are
validated with high score: RAV23 and RAV1552.

```{r}
heatmapTable(val_all, RAVmodel)
```

## Associated MeSH terms
```{r out.height="60%", out.width="60%", fig.width=9, fig.height=9}
drawWordcloud(RAVmodel, 23)
```

```{r out.height="60%", out.width="60%"}
drawWordcloud(RAVmodel, 1552)
```

```{r out.height="60%", out.width="60%", fig.width=9, fig.height=9}
drawWordcloud(RAVmodel, 1387)
```

## Associated studies
```{r}
findStudiesInCluster(RAVmodel, 23, studyTitle = TRUE)
```

```{r}
findStudiesInCluster(RAVmodel, 1552, studyTitle = TRUE)
```

```{r}
findStudiesInCluster(RAVmodel, 1387, studyTitle = TRUE)
```



# Manuscript Figures
## Figure 1
### Figure 1B top-left
```{r eval=FALSE}
x <- calculateScore(dataset, RAVmodel)
top_PC_annot <- c(23,1552,1387,684,338,299,21,312)
Fig1B_topleft <- sampleScoreHeatmap(x[,top_PC_annot],                    
                                    dataName = "E-MTAB-2452", 
                                    modelName = "RAVs for top 8 PCs", 
                                    row_names_gp = 5, column_names_gp = 7, 
                                    cluster_columns = FALSE, cluster_row = FALSE)
Fig1B_topleft
```

```{r echo=FALSE, eval=FALSE}
## Save the Figure 1B top-left
png("Figures/manuscript_Fig1B_topleft.png", width = 1500, height = 1500, res = 300)
Fig1B_topleft
dev.off()
```

```{r 1B_top_left, out.height="60%", out.width="60%", echo=FALSE}
## SampleScore heatmap
img <- readImage("Figures/manuscript_Fig1B_topleft.png")
display(img, method = "raster")
```

### Figure 1B bottom-left
Output from `drawWordcloud(RAVmodel, 1551)`.
```{r echo=FALSE, eval=FALSE}
## Save the Figure 1B bottom-left
png("Figures/manuscript_Fig1B_bottomleft.png", width = 1500, height = 1500, res = 300)
drawWordcloud(RAVmodel, 1551)
dev.off()
```

```{r 1B_bottom_left, echo=FALSE, out.height="60%", out.width="60%"}
## wordcloud
img <- readImage("Figures/manuscript_Fig1B_bottomleft.png")
display(img, method = "raster")
```

### Figure 1B bottom-right
```{r echo=FALSE, eval=FALSE}
a14 <- annotatePC(1:4, val_all, RAVmodel, n = 5, simplify = TRUE, scoreCutoff = 0)
a58 <- annotatePC(5:8, val_all, RAVmodel, n = 5, simplify = TRUE, scoreCutoff = 0)
b <- plotAnnotatedPCA(dataset, RAVmodel, PCnum = c(1,3), val_all = val_all, 
                      scoreCutoff = 0.3, color_by = cellType, 
                      color_lab = "Cell Type", trimed_pathway_len = 45)

library(flextable)
set_flextable_defaults(na_str = "NA")
a14_ft <- flextable(a14) %>% width(., width = 2.5)
a14_ft <- grid::rasterGrob(as_raster(a14_ft))

a58_ft <- flextable(a58) %>% width(., width = 1.5)
a58_ft <- grid::rasterGrob(as_raster(a58_ft))

# ggpubr::ggarrange(a14_ft, a58_ft, labels = c("A", " "), 
#                   hjust = 0.1, vjust = 1.5, align = "hv",
#                   nrow = 2, 
#                   heights = c(0.8, 1),
#                   font.label = list(size = 15)) 
```

```{r eval=FALSE, fig.width=5, fig.height=5}
Fig1B_bottomright <- plotAnnotatedPCA(dataset, RAVmodel, PCnum = c(2,3), 
                                      val_all = val_all, scoreCutoff = 0.3, 
                                      color_by = cellType, 
                                      color_lab = "Cell Type", 
                                      trimed_pathway_len = 45)
Fig1B_bottomright
```

```{r echo=FALSE, eval=FALSE}
## Save the Figure 1B bottom-right
png("Figures/manuscript_Fig1B_bottomright.png", width = 1500, height = 1500, res = 300)
Fig1B_bottomright
dev.off()
```

```{r 1B_bottom_right, echo=FALSE, out.height="80%", out.width="80%"}
## Annotated PCA plot
img <- readImage("Figures/manuscript_Fig1B_bottomright.png")
display(img, method = "raster")
```

## Supplementary Figure 8
PCA result of leukocyte gene expression data (E-MTAB-2452) is displayed in A) a 
table or B) a scatter plot. PCA is done on a centered, but not scaled, input 
dataset by default. Different cutoff parameters for GSEA annotation, such as 
minimum validation score or NES, can be set.

```{r eval=FALSE, fig.width=6.5, fig.height=9.5}
supFig8 <- ggpubr::ggarrange(a14_ft, b, labels = c("a", "b"),
                             nrow = 2, heights = c(1.7, 5), align = "hv") 

supFig8
```

```{r echo=FALSE, eval=FALSE}
## Save the Supplementary Figure 8
png("Figures/manuscript_Sup_Fig8.png", width = 650, height = 950)
supFig8
dev.off()
```

```{r echo=FALSE, out.height="100%", out.width="100%"}
img <- readImage("Figures/manuscript_Sup_Fig8.png")
display(img, method = "raster")
```


# Session Info

<details>
```{r}
sessionInfo()
```
</details>


```{r direct_PCA, collapse=FALSE, echo=FALSE, eval=FALSE}
# PCA
## Directly on E-MTAB-2452
gene_common <- intersect(rownames(RAVmodel), rownames(dataset))
prcomRes <- stats::prcomp(t(dataset[gene_common,]))
loadings <- prcomRes$rotation[, 1:8]

### GSEA on top 8 PCs
library(PLIER)
library(clusterProfiler)

data(canonicalPathways)
data(bloodCellMarkersIRISDMAP)
data(svmMarkers)
allPaths <- combinePaths(canonicalPathways, bloodCellMarkersIRISDMAP, svmMarkers)

source('~/data2/[archive]GenomicSuperSignature/R/gmtToMatrix.R')
term2gene <- matrixToTERM2GENE(allPaths)

res_all <- list()
for (i in seq_len(ncol(loadings))) {
  ## Run GSEA on ranked gene list
  geneList <- loadings[,i]
  geneList <- sort(geneList, decreasing = TRUE)
  res <- GSEA(geneList, TERM2GENE = term2gene, pvalueCutoff = 0.05)
  ## Subset to the most significant pathways
  res <- res[which(res$qvalues == min(res$qvalues)), c("Description", "NES", "pvalue", "qvalues"), drop = FALSE]
  resName <- paste0("PC", i)
  res_all[[resName]] <- res
  res_all[[i]] <- res
}

summary <- list()
for (i in 1:8) {
  annotatedPC <- res_all[[i]]
  topAnnotation <- annotatedPC[order(abs(annotatedPC$NES), decreasing = TRUE),][1:5,]
  rownames(topAnnotation) <- NULL
  summary[[i]] <- topAnnotation
  names(summary)[i] <- paste0("PC", i)
}

# Simple version of the output - only witt the description
simple_summary <- sapply(summary, function(x) x$Description) %>% as.data.frame
simple_summary  
```
